modules/nf-core/rtgtools/bndeval/main.nf

```diff
@@ -0,0 +1,79 @@
+process RTGTOOLS_BNDEVAL {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/rtg-tools:3.12.1--hdfd78af_0':
+        'biocontainers/rtg-tools:3.12.1--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(query_vcf), path(query_vcf_tbi), path(truth_vcf), path(truth_vcf_tbi), path(regions_bed)
+
+    output:
+    tuple val(meta), path("*.tp.vcf.gz")                , emit: tp_vcf
+    tuple val(meta), path("*.tp.vcf.gz.tbi")            , emit: tp_tbi
+    tuple val(meta), path("*.fn.vcf.gz")                , emit: fn_vcf
+    tuple val(meta), path("*.fn.vcf.gz.tbi")            , emit: fn_tbi
+    tuple val(meta), path("*.fp.vcf.gz")                , emit: fp_vcf
+    tuple val(meta), path("*.fp.vcf.gz.tbi")            , emit: fp_tbi
+    tuple val(meta), path("*.tp-baseline.vcf.gz")       , emit: baseline_vcf
+    tuple val(meta), path("*.tp-baseline.vcf.gz.tbi")   , emit: baseline_tbi
+    tuple val(meta), path("*.weighted_roc.tsv.gz")      , emit: weighted_roc
+    tuple val(meta), path("*.summary.txt")              , emit: summary
+    path "versions.yml"                                 , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ""
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def bed_regions = regions_bed ? "--bed-regions=${regions_bed}" : ""
+    def truth_index = truth_vcf_tbi ? "" : "rtg index ${truth_vcf}"
+    def query_index = query_vcf_tbi ? "" : "rtg index ${query_vcf}"
+    def avail_mem = task.memory.toGiga() + "G"
+
+    """
+    ${truth_index}
+    ${query_index}
```

Comment:
I was wondering whether this should be put into an independent module instead. Would there be a reason to add the index building as part of the same module?

Reply:
This was the case for vcfeval, but running it in the same process would make the process faster.  Also, index can also be given as input so this is optional anyways.

Reply:
I agree with @ggabernet that the indexing should go into a different module as you're not piping here [^1].

Even if it's faster it's not really how we do things - particularly if it's optional. Presumably you would reuse the index anyway (so would generate it upfront), and maybe you want the index without running the whole `bndeval` module.

[^1]: https://nf-co.re/docs/guidelines/components/modules#use-of-multi-command-piping

Reply:
But indexing the output is recommended? Here, I am not sure since I never used myself but, indexing with rtgtools might be required for specific callers. https://github.com/nf-core/modules/blob/7b5d8ecb065e4eba0e3dd59653f6011b9470b258/modules/nf-core/rtgtools/vcfeval/main.nf does the same. @nvnieuwk is there any particular reason you made the implementation like that?

Reply:
It is indexing the input, but it is using **rtgtools index** not tabix - so not a different tool.

Reply:
It's not about the tool but rather keeping to the strict structure we aim to have of what goes in a module.

Please remove it as indexing, if not done on the output of an upstream module, should be executed as a standalone module. This means you keep consistency with the majority of other nf-core modules.

 You can ignore @nvnieuwk's 'precedence', he should go back and correct the vcfeval module ;)

Reply:
But not a hill I am willing to die on

Reply:
As the main user of those tools, I would argue that too. Yet, I will change it.